src/test/java/org/broadinstitute/barclay/help/DocumentationGenerationIntegrationTest.java

```diff
@@ -42,46 +49,49 @@
         // set the doclet class
         docArgList.add("-doclet");
         docArgList.add(docletClass.getName());
+        // set the index and output extension
+        docArgList.add("-index-file-extension");
+        docArgList.add(indexFileExtension);
+        docArgList.add("-output-file-extension");
+        docArgList.add(outputFileExtension);
         return docArgList.toArray(new String[]{});
     }
 
     @DataProvider
     public Object[][] getDocGenTestParams() {
         return new Object[][] {
                 // default doclet and templates
-                {HelpDoclet.class, new File("src/main/resources/org/broadinstitute/barclay/helpTemplates/"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/HelpDoclet")},
+                {HelpDoclet.class, new File("src/main/resources/org/broadinstitute/barclay/helpTemplates/"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/HelpDoclet"), "html", "html"},
                 // custom doclet and templates
-                {TestDoclet.class, new File("src/test/resources/org/broadinstitute/barclay/help/templates/TestDoclet"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/TestDoclet")}
+                {TestDoclet.class, new File("src/test/resources/org/broadinstitute/barclay/help/templates/TestDoclet"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/TestDoclet"), "html", "html"}
         };
     }
 
     @Test(dataProvider = "getDocGenTestParams")
-    public void testDocGenRoundTrip(final Class<?> docletClass, final File templatesFolder, final File expectedDir) throws IOException {
+    public void testDocGenRoundTrip(final Class<?> docletClass, final File templatesFolder, final File expectedDir,
+            final String indexFileExtension, final String outputFileExtension) throws IOException {
         // creates a temp output directory
         final File outputDir = Files.createTempDirectory(docletClass.getName()).toAbsolutePath().toFile();
         outputDir.deleteOnExit();
 
         // run javadoc with the the custom doclet
-        com.sun.tools.javadoc.Main.execute(docArgList(docletClass, templatesFolder, outputDir));
+        com.sun.tools.javadoc.Main.execute(docArgList(docletClass, templatesFolder, outputDir, indexFileExtension, outputFileExtension));
+
+        // Compare index files
+        Assert.assertTrue(filesContentsIdentical(new File(expectedDir, "index." + indexFileExtension), new File(outputDir, "index.html")));
```

Comment:
Not your fault, but I just noticed that this reverses the order of expected/actual arguments from the way they're declared by filesContentsIdentical.